File: src/main/java/com/github/snowgooseyk/sscsv/base/Row.java
package com.github.snowgooseyk.sscsv.base;

import java.util.List;

public class Row {

    private final List<?> columns;
    private final int rowNumber;

    public Row(int rowNumber, List<?> columns) {
        this.columns = columns;
        this.rowNumber = rowNumber;
    }

    public int getRowNumber() {
        return rowNumber;
    }

    @SuppressWarnings("unchecked")
    public <T> List<T> getRawColumnValues() {
        return (List<T>) columns;
    }
}

File: src/main/scala/com/github/snowgooseyk/sscsv/Row.scala
package com.github.snowgooseyk.sscsv

import scala.collection.convert.Wrappers.JListWrapper
import com.github.snowgooseyk.sscsv.base.{ Row => JRow }

/**
 * @author snowgooseyk
 */
case class Row(r: JRow) {

  def rownum = r.getRowNumber

  def columns: Seq[String] = JListWrapper(r.getRawColumnValues[String]).toSeq
}
